Build failure: xone with latest kernel

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

boot.kernelPackages = pkgs.linuxPackages_latest;
hardware.xone.enable = true;

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/ffv9zbkg27dx5dzajb6mzpg3cl5mrpcg-source
source root is /build/source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j8 SHELL=/nix/store/ih68ar79msmj0496pgld4r3vqfr7bbin-bash-5.2p37/bin/bash -C /nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modu>
make: Entering directory '/nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modules/6.16.0/build'
make[1]: Entering directory '/build/source'
  CC [M]  transport/wired.o
  CC [M]  transport/dongle.o
  CC [M]  transport/mt76.o
  CC [M]  bus/bus.o
  CC [M]  bus/protocol.o
  CC [M]  auth/auth.o
  CC [M]  auth/crypto.o
  CC [M]  driver/common.o
  CC [M]  driver/gamepad.o
  CC [M]  driver/headset.o
  CC [M]  driver/chatpad.o
  CC [M]  driver/madcatz_strat.o
  CC [M]  driver/madcatz_glam.o
  CC [M]  driver/pdp_jaguar.o
transport/dongle.c: In function 'xone_dongle_init':
transport/dongle.c:892:46: warning: too many arguments for format [-Wformat-extra-args]
  892 |                         snprintf(fwname, 25, "xow_dongle.bin",
      |                                              ^~~~~~~~~~~~~~~~
driver/gamepad.c: In function 'gip_gamepad_send_rumble':
driver/gamepad.c:102:45: error: implicit declaration of function 'from_timer'; did you mean 'mod_timer'? [-Wimplicit-function-declaration]
  102 |         struct gip_gamepad_rumble *rumble = from_timer(rumble, timer, timer);
      |                                             ^~~~~~~~~~
      |                                             mod_timer
driver/gamepad.c:102:45: error: initialization of 'struct gip_gamepad_rumble *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
make[3]: *** [/nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modules/6.16.0/source/scripts/Makefile.build:287: driver/gamepad.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [/nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:2003: .] Error 2
make[1]: *** [/nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/build/source'
make: *** [/nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
make: Leaving directory '/nix/store/l837ccxwg988mx8nsgy54x35w79v5f3q-linux-6.16-dev/lib/modules/6.16.0/build'
```

### Additional context

There is a release to fix this issue on the xone repo, and it is live on the unstable channel. 

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.8, NixOS, 25.05 (Warbler), 25.05.20250726.13e8d35`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/vjk8x60zj28wj5n07276s0cj7594bncn-source`

### Notify maintainers

@rhysmdnz @fxzzi

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/xone/default.nix
+++ b/pkgs/os-specific/linux/xone/default.nix
@@ -6,13 +6,13 @@
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "xone";
-  version = "0.3.2";
+  version = "0.3.4";
 
   src = fetchFromGitHub {
     owner = "dlundqvist";
     repo = "xone";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-sX+oN7H0Okl+3yYwbIFoHU8q2pOcMHQcRqQkmnMXSps=";
+    hash = "sha256-8ljHKVo+mdzm8CJpf0y1xXjzhQF/qQotMmxDn7G/EVk=";
   };
 
   setSourceRoot = ''

--- a/pkgs/os-specific/linux/xone/default.nix
+++ b/pkgs/os-specific/linux/xone/default.nix
@@ -6,13 +6,13 @@
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "xone";
-  version = "0.3.1";
+  version = "0.3.2";
 
   src = fetchFromGitHub {
     owner = "dlundqvist";
     repo = "xone";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-qMZlQgAe5vB5zfuhyK7EBxIwfhnC5MvnF/qr3BGnDms=";
+    hash = "sha256-sX+oN7H0Okl+3yYwbIFoHU8q2pOcMHQcRqQkmnMXSps=";
   };
 
   setSourceRoot = ''

--- a/pkgs/os-specific/linux/xone/default.nix
+++ b/pkgs/os-specific/linux/xone/default.nix
@@ -6,13 +6,13 @@
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "xone";
-  version = "0.3.4";
+  version = "0.3.5";
 
   src = fetchFromGitHub {
     owner = "dlundqvist";
     repo = "xone";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-8ljHKVo+mdzm8CJpf0y1xXjzhQF/qQotMmxDn7G/EVk=";
+    hash = "sha256-O+SdUx5wFIT4mLUO5awFp9IKmtnRhMtBLEHaRXRBv2Y=";
   };
 
   setSourceRoot = ''

